spec/spec_helper.rb

```diff
@@ -61,6 +61,26 @@
   # triggering implicit auto-inclusion in groups with matching metadata.
   config.shared_context_metadata_behavior = :apply_to_host_groups
 
+  config.before(:suite) do
```

Comment:
Could you explain how those database cleaning works?

there is a `before :suite` that it cleans with `:deletion`

then there is this `:each` but it sets a strategy...

then we have the same but with an additional `:js => true`

Reply:
- The `config.before(:suite)` with a `clean_with(:deletion)` will make sure the database is clean before the whole suite starts in case there are any leftovers from a previous run.

- Setting the strategy will dictate what will be DB Cleaner's approach to revert the test cases, in this example, it will open new deletions on each test case and delete it once it’s finished.

- The `:js => true` is to include the javascript-enabled tests.

PS: This is what I found digging through some forums and articles, please do take this with a grain of salt since I've recently starting looking into DatabaseCleaner

Reply:
That's mainly it